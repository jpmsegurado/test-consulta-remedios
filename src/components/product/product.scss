@import "~@/variables.scss";

.product {

  .image {
    background-color: $product-bg;
    padding: 25px 41px;
    width: 100%;
    box-sizing: border-box;
    border-radius: 3px;
    display: flex;
    justify-content: center;
    align-items: center;

    img {
      object-fit: contain;
      width: 180px;
      height: 180px;
    }
  }

  &:hover {
    cursor: pointer;
    .info {
      opacity: 0;
    }

    .action {
      opacity: 1;
    }
  }

  .info {
    opacity: 1;
    transition: .5s;
    > p {
      width: 100%;
      display: block;
      font-size: 16px;
      text-align: center;
      margin: 0;
  
      &:first-of-type {
        margin-top: 13px;
        color: $product-name;
      }
  
      &:last-of-type {
        margin-top: 2px;
        color: $product-price;
        font-weight: bold;
      }
    }
  }

  .action {
    opacity: 0;
    outline: none;
    position: relative;
    top: -33px;
    transition: .5s;

    button {
      display: block;
      width: 100%;
      background-color: transparent;
      color: $product-price;
      border: 0;
      font-size: 16px;
      font-weight: bold;
      outline: none;
      cursor: pointer;
    }
  }
}