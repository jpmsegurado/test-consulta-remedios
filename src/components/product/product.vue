<template>
  <div class="product">
    <div class="image">
      <img :src="require(`@/assets/${productInfo.image}`)" alt="fifa 18" />
    </div>
    <div class="info">
      <p>{{ productInfo.name }}</p>
      <p>{{ formatMoney(productInfo.price) }}</p>
    </div>

    <div class="action">
      <button>
        adicionar ao carrinho
      </button>
    </div>
  </div>
</template>

<script>
export default {
  name: "product",
  props: {
    productInfo: {
      type: Object,
      default: () => ({})
    }
  }
};
</script>

<style lang="scss" scoped src="./product.scss" />
