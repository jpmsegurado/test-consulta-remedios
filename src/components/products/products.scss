.products {
  width: 100%;
  display: flex;
  flex-wrap: wrap;

  > .product-wrapper {
    width: calc(100% / 3);
    padding: 15px;
    box-sizing: border-box;
  }
}