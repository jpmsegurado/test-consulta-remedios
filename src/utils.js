export function formatMoney(value, decimal = 2) {
  const formatter = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
    minimumFractionDigits: decimal
  });

  return formatter.format(value);
}
