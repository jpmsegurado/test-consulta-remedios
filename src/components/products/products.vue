<template>
  <div class="products">
    <div class="product-wrapper" v-for="product in products" :key="product.id">
      <product :product-info="product" />
    </div>
  </div>
</template>

<script>
import product from "../product/product";
import { mapGetters, mapActions } from "vuex";

export default {
  components: {
    product
  },
  computed: {
    ...mapGetters({
      products: "PRODUCT/ALL"
    })
  },
  methods: {
    ...mapActions({
      loadProducts: "PRODUCT/LOAD_PRODUCTS"
    })
  },
  mounted() {
    this.loadProducts();
  }
};
</script>

<style lang="scss" scoped src="./products.scss" />
