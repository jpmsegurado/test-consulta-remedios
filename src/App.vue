<template>
  <div id="app">
    <div class="row">
      <div class="products-wrapper">
        <h2>Games</h2>
        <mySelect v-model="selected" :value="selected" :list="options" />
        <products :selectedOption="options[selected]" />
      </div>

      <div class="shopping-cart-wrapper">
        <shopping-cart />
      </div>
    </div>
  </div>
</template>

<script>
import products from "./components/products/products";
import shoppingCart from "./components/shopping-cart/shopping-cart";
import mySelect from "./components/select/select";

export default {
  name: "app",
  components: {
    products,
    shoppingCart,
    mySelect
  },
  data() {
    return {
      options: [
        { label: "Mais populares", prop: "score" },
        { label: "Mais baratos", prop: "price" },
        { label: "Ordem Alfab√©tica", prop: "name" }
      ],
      selected: 0
    };
  }
};
</script>

<style lang="scss">
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
body {
  padding: 0;
  margin: 0;
}
#app {
  font-family: "Open Sans";
  padding: 42px 70px;

  > .row {
    display: flex;
    width: 100%;
    flex-wrap: wrap;

    .products-wrapper {
      width: 75%;

      > h2 {
        font-size: 48px;
        margin-top: 0;
        margin-bottom: 44px;
        width: calc(100% - 277px);
        display: inline-block;
      }

      > .select {
        width: 262px;
        margin-right: 15px;
      }
    }

    .shopping-cart-wrapper {
      width: 25%;
    }
  }
}
</style>
