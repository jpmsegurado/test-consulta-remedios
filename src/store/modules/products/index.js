import products from "../../../assets/products.json";

const state = {
  products: []
};

const getters = {
  ALL: state => state.products
};

const actions = {
  LOAD_PRODUCTS: ({ commit }) => {
    return Promise.resolve(commit("SET_PRODUCTS", products));
  }
};

const mutations = {
  SET_PRODUCTS: (state, products) => {
    state.products = products;
  }
};

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
};
