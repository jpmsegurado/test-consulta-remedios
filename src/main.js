import Vue from "vue";
import App from "./App.vue";
import store from "./store/store";
import { formatMoney } from "./utils";

Vue.config.productionTip = false;

Vue.prototype.formatMoney = formatMoney;

new Vue({
  store,
  render: h => h(App)
}).$mount("#app");
