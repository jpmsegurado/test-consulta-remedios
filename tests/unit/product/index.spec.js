import { shallowMount } from "@vue/test-utils";
import product from "@/components/product/product.vue";
import products from "@/assets/products.json";
import { formatMoney } from "../../../src/utils";

describe("product.vue", () => {
  it("render product info normally", () => {
    const productInfo = products[0];
    const wrapper = shallowMount(product, {
      propsData: { productInfo },
      mocks: {
        formatMoney
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
