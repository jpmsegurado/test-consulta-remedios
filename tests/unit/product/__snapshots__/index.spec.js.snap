// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product.vue render product info normally 1`] = `
<div class="product">
  <div class="image"><img src="" alt="fifa 18"></div>
  <div class="info">
    <p>Super Mario Odyssey</p>
    <p>R$&nbsp;197.88</p>
  </div>
  <div class="action"><button>
      adicionar ao carrinho
    </button></div>
</div>
`;
